kyo-prelude/shared/src/test/scala/kyo/ChoiceTest.scala

```diff
@@ -97,4 +97,51 @@ class ChoiceTest extends Test:
         end try
         ()
     }
+
+    "interaction with collection operations" - {
+        "foreach" in {
+            val result = Choice.run(
+                Kyo.foreach(List("x", "y")) { str =>
+                    Choice.get(List(true, false)).map(b =>
```

Comment:
Note how `Choice` will branch the computation while the collection operation is still being processed. If `foreach` used a mutable buffer, `Choice` calling the continuation multiple times would produce inconsistent results.